addons/binding/org.openhab.binding.roomba/src/main/java/org/openhab/binding/roomba/handler/RoombaHandler.java

```diff
@@ -0,0 +1,184 @@
+ * sent to one of the channels.
+ *
+ * @author Stephen Liang - Initial contribution
+ */
+public class RoombaHandler extends BaseThingHandler {
+    private RoombaService roombaService;
+
+    private Logger logger = LoggerFactory.getLogger(RoombaHandler.class);
+
+    public RoombaHandler(Thing thing) {
+        super(thing);
+
+        RoombaConfiguration config = getConfigAs(RoombaConfiguration.class);
+
+        // Keeping this flexible, iRobot may decide to change how the APIs are done for future Roomba products.
+        try {
+            switch (thing.getThingTypeUID().getId()) {
+                case ROOMBA_980_SKU:
+                    this.roombaService = new Roomba900Service(config.ipAddress, config.password);
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                            "Cannot support this thing yet: " + thing.getThingTypeUID().getId());
+            }
+
+        } catch (Exception e) {
+            logger.error("Failed to start roomba 980 service", e);
+        }
+    }
+
+    /**
+     * Handles the command for a given channel
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("received command for channel uid: {}, command: {}", channelUID, command);
+
+        switch (channelUID.getId()) {
+            case CHANNEL_START:
+                handleStart(command);
+                break;
+            case CHANNEL_PAUSE:
+                handlePause(command);
+                break;
+            case CHANNEL_STOP:
+                handleStop(command);
+                break;
+            case CHANNEL_RESUME:
+                handleResume(command);
+                break;
+            case CHANNEL_DOCK:
+                handleDock(command);
+                break;
+            default:
+                logger.debug("Received command {} for unknown channel: {}", command, channelUID);
+                break;
+        }
+    }
+
+    /**
+     * Initializes this handler and ensures that the Roomba is available for commands.
+     */
+    @Override
+    public void initialize() {
+        try {
+            if (roombaService.isOk()) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Failed to get the Roomba's current mission.");
+            }
+        } catch (RoombaCommunicationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Handles the start channel. This channel starts the Roomba's cleaning process
+     *
+     * @param command The command being handled
+     */
+    private void handleStart(Command command) {
+        try {
+            if (command instanceof OnOffType) {
```

Comment:
if you call handleStart with OnOffType.OFF it starts the service is that completely expected?